src/Lightning.ts

```diff
@@ -638,8 +680,29 @@ export const LightningMixin = (superclass) => class extends superclass {
     })
   }
 
-  async updatePendingInvoice({ hash, expired = false, lock }) {
-    let invoice
+  // return whether the invoice has been paid of not
+  async updatePendingInvoice({ hash, lock, pubkey }: {hash: string, lock: any, pubkey?: string}): Promise<boolean> {
+    let invoice, pubkey_
+    
+    // if a pubkey has been provided, it means the invoice has not been set as paid in mongodb
+    // so not need for a round back trip to mongodb
+    if (!pubkey) {
+      let paid
+      ({ pubkey: pubkey_, paid } = await InvoiceUser.findOne({ _id: hash }));
```

Comment:
Why not fetch everything here and remove the call on #L733?

Reply:
good question.

the idea is that the first check here is to see whether the invoice has been paid. we only do it if the pubkey is not passed as an argument, which is when it's called from graphql directly (from updatePendingInvoice or noauthUpdatePendingInvoice).

if it is, we can just return the function here. the benefits of doing that is that we only lock the account if the invoice has been paid but is not yet updated in medici. 

if we wanted to have a single call to InvoiceUser, we should basically have to lock the account on every request, which is not ideal. if we don't use a lock, we could have a race condition and payment could be validated multiple time. 

does that make sense?

Reply:
Yep